extensions/modules/android-utils/android-utils.lcb

```diff
@@ -232,4 +232,109 @@ public handler AndroidHasPermission(in pPermission as String) returns Boolean
 	return tHasPermission
 end handler
 
+public handler type OnPauseHandler() returns nothing
+public handler type OnResumeHandler() returns nothing
+
+__safe foreign handler _JNI_Engine_LifecycleListener(in pListener as Array) returns JObject \
+    binds to "java:com.runrev.android.Engine$LifecycleListener>interface()"
+__safe foreign handler _JNI_Engine_RegisterLifecycleListener(in pSelf as JObject, in pListener as JObject) returns JBoolean \
+    binds to "java:com.runrev.android.Engine>registerLifecycleListener(Lcom/runrev/android/Engine$LifecycleListener;)Z"
+__safe foreign handler _JNI_Engine_UnregisterLifecycleListener(in pSelf as JObject, in pListener as JObject) returns JBoolean \
+    binds to "java:com.runrev.android.Engine>unregisterLifecycleListener(Lcom/runrev/android/Engine$LifecycleListener;)Z"
+
+/**
+Summary:
+Register handlers to be called when on application lifecycle events.
+
+Parameters:
+pPauseHandler: The handler to be called when the application is paused
+pResumeHandler: The handler to be called when the application is resumed
+
+Returns:
+A listener object that wraps the registered handlers.
+
+Example:
+    private variable mLifecycleListener as optional JObject
+
+    public handler OnOpen() returns nothing
+        put AndroidRegisterLifecycleListener(OnPause, OnResume) into mLifecycleListener
+    end handler
+
+    public handler OnClose() returns nothing
+        AndroidUnregisterLifecycleListener(mLifecycleListener)
+        put nothing into mLifecycleListener
+    end handler
+
+    public handler OnPause() returns nothing
```

Comment:
I suggest that we recommend people *do not* use public `OnPause` and `OnResume` as the names - and instead use more specific naming using private handlers. This is so we can (in the future) add automatic registration of such lifecycle handlers (i.e. have widget events which do the same thing without having to register / deregister).